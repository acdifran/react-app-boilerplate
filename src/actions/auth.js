import { firebase, googleAuthProvider } from '../firebase/firebase';

export const startLogin = () => () =>
  firebase.auth().signInWithPopup(googleAuthProvider);

export const startLogout = () => () => firebase.auth().signOut();

export const login = uid => ({
  type: 'LOGIN',
  uid
});

export const logout = () => ({
  type: 'LOGOUT'
});
