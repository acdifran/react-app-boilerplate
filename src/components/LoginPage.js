import React from "react";
import { startLogin } from "../actions/auth";

export default () => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">React App</h1>
      <p>Tagline for application</p>
      <button className="button" onClick={startLogin()}>
        Login with Google
      </button>
    </div>
  </div>
);
