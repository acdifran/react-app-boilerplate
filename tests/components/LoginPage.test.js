import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from '../../src/components/LoginPage';
import * as auth from '../../src/actions/auth';

test('should render login page correctly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
  const mockLogin = jest.spyOn(auth, 'startLogin');
  const wrapper = shallow(<LoginPage />);
  wrapper.find('button').simulate('click');
  expect(mockLogin).toHaveBeenCalled();
});
