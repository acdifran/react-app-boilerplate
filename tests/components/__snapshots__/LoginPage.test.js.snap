// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      React App
    </h1>
    <p>
      Tagline for application
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
